app/views/linkers/index.html.erb

```diff
@@ -48,25 +48,26 @@ function readCSVFile() {
 
     var lines = reader.result.split('\n');
 
-    for(var i = 0; i < lines.length; i+=1) {
+    lines.forEach(function(line) {
 
       var new_row = document.createElement('tr');
-      var new_cell = document.createElement('td'); new_cell.appendChild(document.createTextNode(lines[i]));
+      var new_cell = document.createElement('td'); new_cell.appendChild(document.createTextNode(line));
       new_row.appendChild(new_cell);
 
       var xhttp = new XMLHttpRequest();
-      xhttp.open("GET", "http://linkfixer.herokuapp.com/ping_url?url_string=" + encodeURIComponent(lines[i]) + "&search_term=" + encodeURIComponent(document.getElementById('search_term').value), false);
+      xhttp.open("GET", "ping_url?url_string=" + encodeURIComponent(line) + "&search_term=" + encodeURIComponent(document.getElementById('search_term').value));
+      xhttp.onreadystatechange = function() {
```

Comment:
Since we're no longer synchronous, you can't simply call `xhttp.send()` and then check its `readyState` and `status` before appending the new rows. Instead, you have to give it an `onreadystatechange` callback for it to call once the request has successfully completed (since it's now asynchronous).